### PR TITLE
Add per-OS execution duration to usage table

### PR DESCRIPTION
**Pull request by bduffany**
Also add tests that make it ~impossible to forget to update the usage tracker when new UsageCounts fields are added.

---

**Version bump**: Patch 

**Related issues**: https://github.com/buildbuddy-io/buildbuddy-internal/issues/1063

### CODE REVIEW

File: enterprise/server/remote_execution/execution_server/execution_server.go [hunk 1]
```diff
@@ -633,14 +634,17 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
 	}
 }
 
-func (s *ExecutionServer) updateRouter(ctx context.Context, taskID string, executeResponse *repb.ExecuteResponse) error {
```
Comment (tylerwilliams) [inline enterprise/server/remote_execution/execution_server/execution_server.go:636]: nit: maybe these two things (updateRouter and updateTaskAccounting) don't really belong together. Why not just make another for task accounting ?

**Review by tylerwilliams — CHANGES_REQUESTED** [review]
mostly looks good, couple of minor things

File: enterprise/server/remote_execution/execution_server/execution_server.go [hunk 2]
```diff
@@ -633,14 +634,17 @@ func (s *ExecutionServer) PublishOperation(stream repb.Execution_PublishOperatio
...
+func (s *ExecutionServer) updateTaskAccounting(ctx context.Context, taskID string, executeResponse *repb.ExecuteResponse) error {
 	router := s.env.GetTaskRouter()
-	if router == nil {
-		return nil
-	}
 	// If the result was served from cache, nothing was actually executed, so no
 	// need to update the task router.
-	if executeResponse.GetCachedResult() {
+	shouldUpdateRouter := router != nil && !executeResponse.GetCachedResult()
```
[the remark this answers is not in the document]
Comment (bduffany) [reply enterprise/server/remote_execution/execution_server/execution_server.go]: cleaned it up a bit 🦂🦂🦂

**Review by tylerwilliams — APPROVED** [review]
LGTM

Thanks, this looks a lot better!

**Pull request merged**